[ISSUE #13824] fix(auth): add AI gRPC resource parser and enable auth for MCP/Agent requests

nihongye opened the pull request:
## What is the purpose of the change

Fix a security gap in Nacos 3.x where AI-related gRPC requests (e.g., `AbstractMcpRequest`, `AbstractAgentRequest`) are not properly authenticated, even when authentication is enabled. This change ensures that all incoming AI protocol requests go through the standard auth pipeline by adding a dedicated resource parser and registering it in the gRPC auth service.

Without this fix, unauthenticated clients could potentially access AI services if the gRPC port is exposed — a risk especially in multi-tenant or public-facing environments.

Related Issue: #13824

## Brief changelog

- Added `AiGrpcResourceParser` to parse namespace, group, and resource name from `AbstractMcpRequest` and `AbstractAgentRequest`.
- Registered `AiGrpcResourceParser` under `SignType.AI` in `GrpcProtocolAuthService`.
- Enhanced `AiGrpcClient` to initialize `SecurityProxy` with periodic token refresh (fix missing login scheduling).
- Added comprehensive unit tests:
  - `GrpcProtocolAuthServiceTest`: added test cases for MCP and Agent request parsing.
  - `AiGrpcResourceParserTest`: implemented parameterized tests covering full context, null namespace, and empty name scenarios.


## Verifying this change

1. **Unit Tests**: All new and existing tests pass:
   ```bash
   mvn clean install -DskipITs
    ```
✅ GrpcProtocolAuthServiceTest.testParseResourceWithMcpType
✅ GrpcProtocolAuthServiceTest.testParseResourceWithAgentType
✅ AiGrpcResourceParserTest.testParse (parameterized across 9 cases)


Follow this checklist to help us incorporate your contribution quickly and easily:

* [x] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [x] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [x] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/alibaba/nacos?pullRequest=13827) <br/>All committers have signed the CLA.

nihongye committed:
fix(auth): add AI gRPC resource parser and enable auth for MCP/Agent requests

Resolves #13824

This commit adds support for authentication of AI-related gRPC requests
(AbstractMcpRequest and AbstractAgentRequest) in the Nacos auth module.

Key changes:
- Implement AiGrpcResourceParser to extract namespace, group and resource name
  from AI protocol requests.
- Register AiGrpcResourceParser under SignType.AI in GrpcProtocolAuthService.
- Add comprehensive unit tests using parameterized testing to cover both
  MCP and Agent request types, including edge cases (null/empty fields).
- Fix missing security token refresh in AiGrpcClient by initializing
  SecurityProxy with scheduled login task.

Ensures that all incoming AI gRPC requests are properly authenticated
when security is enabled, closing a previous authorization gap.

wuyfee commented:
$\color{red}{FAILURE}$
[DETAILS](https://github.com/alibaba/nacos/actions/runs/17815520507)
 ✅ - docker:   success 
 ❌ - deploy (standalone & cluster & standalone_auth):   failure 
 ❌ - e2e-java-test (standalone & cluster & standalone_auth):   skipped 
 ❌ - e2e-go-test (standalone & cluster):   skipped 
 ❌ - e2e-cpp-test (standalone & cluster):   skipped 
 ❌ - e2e-csharp-test (standalone & cluster):   skipped 
 ❌ - e2e-nodejs-test (standalone & cluster):   skipped 
 ❌ - e2e-python-test (standalone & cluster):   skipped 
 ✅ - clean (standalone & cluster & standalone_auth):   success

wuyfee commented:
$\color{red}{FAILURE}$
[DETAILS](https://github.com/alibaba/nacos/actions/runs/17815696265)
 ✅ - docker:   success 
 ❌ - deploy (standalone & cluster & standalone_auth):   failure 
 ❌ - e2e-java-test (standalone & cluster & standalone_auth):   skipped 
 ❌ - e2e-go-test (standalone & cluster):   skipped 
 ❌ - e2e-cpp-test (standalone & cluster):   skipped 
 ❌ - e2e-csharp-test (standalone & cluster):   skipped 
 ❌ - e2e-nodejs-test (standalone & cluster):   skipped 
 ❌ - e2e-python-test (standalone & cluster):   skipped 
 ✅ - clean (standalone & cluster & standalone_auth):   success
